File: Javascript/traversy-js-challenges-main/04-recursion/10-permutations/permutations.js
function permutations(string) {
    let arrayOfPermutations = [];
    if (string === '') {
        arrayOfPermutations.push(string);
        return arrayOfPermutations;
    }
    for (let i =0; i < string.length; i++) {
        const currentChar = string[i];
        const restOfString = string.slice(0, i) + string.slice(i + 1);
        const subPermutations = permutations(restOfString);
        for (let j = 0; j < subPermutations.length; j++) {
            arrayOfPermutations.push(currentChar + subPermutations[j]);
        }
    }
    return arrayOfPermutations;
}

module.exports = permutations;

File: Python/RecursionPractice/permutations/permutations.py

class Permutations:

    def get_list_of_permutations(self, string):
        permutations = []
        if string == "":
            permutations.append(string)
            return permutations
        for i in range(len(string)):
            current_char = string[i]
            rest_of_string = string[0:i] + string[i + 1:]
            sub_permutations = self.get_list_of_permutations(rest_of_string)
            for j in range(len(sub_permutations)):
                permutations.append(current_char + sub_permutations[j])
        return permutations
